x-pack/legacy/plugins/uptime/public/components/functional/location_map/location_status_tags.tsx

```diff
@@ -20,54 +27,97 @@ const BadgeItem = styled.div`
 
 const TagContainer = styled.div`
   padding: 10px;
-  max-height: 200px;
+  max-height: 229px;
   overflow: hidden;
 `;
 
+const OtherDiv = styled.div`
+  padding-left: 18px;
+`;
+
 interface Props {
   locations: MonitorLocation[];
 }
 
+interface StatusTag {
+  label: string;
+  timestamp: number;
+}
+
 export const LocationStatusTags = ({ locations }: Props) => {
   const {
     colors: { gray, danger },
   } = useContext(UptimeSettingsContext);
 
-  const upLocs: string[] = [];
-  const downLocs: string[] = [];
+  const upLocations: StatusTag[] = [];
+  const downLocations: StatusTag[] = [];
 
   locations.forEach((item: any) => {
     if (item.summary.down === 0) {
-      upLocs.push(item.geo.name);
+      upLocations.push({ label: item.geo.name, timestamp: new Date(item.timestamp).valueOf() });
     } else {
-      downLocs.push(item.geo.name);
+      downLocations.push({ label: item.geo.name, timestamp: new Date(item.timestamp).valueOf() });
     }
   });
 
+  // Sort by recent timestamp
+  upLocations.sort((a, b) => {
+    return a.timestamp < b.timestamp ? 1 : b.timestamp < a.timestamp ? -1 : 0;
+  });
+
+  moment.locale('en', {
+    relativeTime: {
+      future: 'in %s',
+      past: '%s ago',
+      s: 'few sec',
+      ss: '%s sec',
+      m: 'a min',
+      mm: '%d min',
+      h: 'an hour',
+      hh: '%d hour',
+      d: 'a day',
+      dd: '%d day',
+      M: 'a month',
+      MM: '%d Mon',
+      y: 'a year',
+      yy: '%d Year',
+    },
+  });
+
+  const tagLabel = (item: StatusTag, ind: number, color: string) => (
+    <BadgeItem key={ind}>
+      <EuiBadge color={color}>
+        <EuiText size="m">
+          <TextStyle>{item.label}</TextStyle>
+        </EuiText>
+      </EuiBadge>
+      <TimeStampSpan>
+        <EuiText color="subdued">{moment(item.timestamp).fromNow()}</EuiText>
+      </TimeStampSpan>
+    </BadgeItem>
+  );
+
   return (
-    <TagContainer>
-      <span>
-        {downLocs.map((item, ind) => (
-          <BadgeItem key={ind}>
-            <EuiBadge color={danger}>
-              <EuiText size="m">
-                <TextStyle>{item}</TextStyle>
-              </EuiText>
-            </EuiBadge>
-          </BadgeItem>
-        ))}
-      </span>
-      <span>
-        {upLocs.map((item, ind) => (
-          <BadgeItem key={ind}>
-            <EuiBadge color={gray}>
-              <EuiText size="m">
-                <TextStyle>{item}</TextStyle>
-              </EuiText>
-            </EuiBadge>
-          </BadgeItem>
-        ))}
-      </span>
-    </TagContainer>
+    <>
+      <TagContainer>
+        <span>{downLocations.map((item, ind) => tagLabel(item, ind, danger))}</span>
+        <span>{upLocations.map((item, ind) => tagLabel(item, ind, gray))}</span>
+      </TagContainer>
+      {locations.length > 7 && (
+        <OtherDiv>
```

Comment:
WDYT about naming this `OtherLocationsDiv`? When I read it I was a little confused and had to look at the code around for additional context.

Reply:
done